### PR TITLE
Add library documentation

### PR DESCRIPTION
**Pull request by alexhauser**
Issue #22.

**Description:**
Add documentation to the library part of this project.

**Notes:**
This is a work in progress, please only merge once the WIP label is removed.

### CODE REVIEW

**Comment by alexhauser** [comment]
@josegomez, please feel free to take a quick look at the current state of my documentation effort and let me know if you have any comments or improvement suggestions.

**Comment by josegomez** [comment]
Very nice!

**Comment by alexhauser** [comment]
@josegomez, this was a little more work than I've expected, man did you write a lot of code during these last few weeks 😁.

This adds documentation to each and every publicly exposed bit of the library project, and some of the internal parts, too. I hope I've got everything correct, especially on the protocol part where I had to figure out a lot myself reading the code before I was able to write the documentation. If you could find some time to double-check those sometime, that would be great.

I've also stumbled upon some inconsistencies and mabye some bugs, too, but I did not want to turn my documentation branch into a bugfix branch, so I will address those later in a separate PR.

I'm now waiting for the CI checks to finish, and if they pass, I guess this one can be merged.

Cheers, Alex

**Comment by josegomez** [comment]
Beautiful man thank you so much! Yes it's a lot of stuff haha!

**Comment by alexhauser** [comment]
@josegomez, I almost forgot to mention that this PR also contains 3 naming refactorings, two instances where member varables were named incorrectly in my opinion, and one typo.

Please see commits:
https://github.com/sqrldev/SQRLDotNetClient/pull/21/commits/0e06ad4f7553b7fece2b557720742c30fece0d4e
https://github.com/sqrldev/SQRLDotNetClient/pull/21/commits/eedc64e8da379fd1b07f3fd2981bab1722ec11c3
https://github.com/sqrldev/SQRLDotNetClient/pull/21/commits/537d82ccd50cc75b337f2c1b1650b181da32cff0

I hope that's also fine with you.

**Comment by alexhauser** [comment]
Checks have passed, so this is ready to be merged as far as I'm concerned.

**Review by josegomez — APPROVED** [review]
Awesome work thanks a lot!

File: SQRLUtilsLib/SQRL.cs [hunk 1]
```diff
@@ -953,67 +1008,79 @@ public SQRLServerResponse GenerateQueryCommand(Uri sqrl, KeyPair siteKP, SQRLOpt
...
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="sqrl">The SQRL server URI.</param>
+        /// <param name="siteKP">The site-specific ECDH public-private key pair.</param>
+        /// <param name="encodedServerMessage">The base64_url-encoded contents for the server parameter.</param>
+        /// <param name="ilk">The Identity Lock Key (ILK).</param>
+        /// <param name="ursKey">The Unlock Request Signing Key (URSK).</param>
+        /// <param name="priorKey">The Previous Identity Key (PIDK) key pair to be appended to the client message.</param>
+        /// <param name="opts">Optional SQRL options to be appended to the client message.</param>
```
Comment (josegomez) [inline SQRLUtilsLib/SQRL.cs:1042]: This one does an Ident but signs it with the old URS and sends new SUK and VUK effectively replacing the old identity with a new rekeyed one

**Review by josegomez — COMMENTED** [review]
All good!

File: SQRLUtilsLib/SQRL.cs (hunk 1 above)
Comment (alexhauser) [reply SQRLUtilsLib/SQRL.cs:1042]: Sorry, @josegomez, I saw your approval and thought you want me to merge the PR myself.
I'll still add that one to a later PR.

Comment (josegomez) [reply SQRLUtilsLib/SQRL.cs:1042]: No worries, I saw this little thing after I looked at it again, no bigdeal

Comment (josegomez) [reply SQRLUtilsLib/SQRL.cs:1042]: and yes fee free to merge anything as long as it passes the tests etc I have no issues with that

**Pull request merged**